### DISCUSSION TITLE
Models building one extra time when using search tuner

### DISCUSSION THREAD

**Discussion by nemesgyadam** [keras-team/keras-tuner#514] [Q&A]
I dont understand why is the build_model function called in the __init__ of HyperModel?
I would like to use different generators to each model of the search place. Which is problematic when the output of the generators are different.
When I initializing my HyperModel class like
`hypermodel = MyHyperModel(conf=conf, conf_objects=conf_objects)
`
it build a model and also initalize my tf.data generators.
And when I run
`tuner.search`
new models and generators initailzed for every elemnt of search place
(which is the desired behavior)

**Comment by haifeng-jin** [comment]
It is called mainly for the tuner to fetch all the hyperparameters to get a sense of the search space.
